Toolkit/ArcGISToolkit/TemplatePickerViewController.swift

```diff
@@ -179,8 +179,12 @@ public class TemplatePickerViewController: TableViewController {
                     // update info with swatch
                     infos[index].swatch = image
                     
-                    // The size of the table view is still not finalized at this point, so we can't use `reloadRows(...)`.
-                    self.tableView.reloadData()
+                    let indexPath = self.indexPath(for: info)
+                    
+                    // The size of the table view is still not finalized at this point, so we check to make sure the section number of `indexPath` does exist in the `tableView`.
+                    if indexPath.section < self.tableView.numberOfSections {
```

Comment:
Won't we also want to check to make sure the number of rows < those in the section?

Reply:
Yep and do you mean this `indexPath.row < self.tableView.numberOfRows(inSection: indexPath.section)`

Reply:
Yup.  It could be that the section was created, but no rows were populated yet.  This all seems like a timing issue to me.

Reply:
Yep it seems that way.

Reply:
Everything should be happening on the main thread, including the response from createSwatch. So I'm thinking it's just that some swatches in this repro case came back immediately before the next iteration of the run loop when table view has a chance to ask the datasource how many sections. So, if the section is there I'm thinking the row would be as well. But it probably doesn't hurt to check.